### PR TITLE
🧪 [testing improvement] Add tests for useUserSettings hook

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** The `useUserSettings` custom hook lacked unit tests, representing a testing gap for user settings functionality in the frontend.

📊 **Coverage:** This PR introduces tests covering:
- Successful fetching of user settings via `useUserSettings`.
- Successful updating of user settings via `useUpdateUserSettingsMutation`, including verification of the required `queryClient.invalidateQueries` behavior.
- Expected API error scenarios (both custom error messages and fallback messages).
- Edge cases where the `userEmail` or session might be missing.

✨ **Result:** Test coverage for `packages/web-app-vercel/lib/hooks/useUserSettings.ts` has increased from 0% to 100%. The code behaves fully as expected.

---
*PR created automatically by Jules for task [888349491552320431](https://jules.google.com/task/888349491552320431) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

このPRは `useUserSettings` カスタムフックのユニットテストを新規追加し、カバレッジを0%から100%に引き上げます。

- `useUserSettings` の正常取得・APIエラー（カスタム/デフォルトメッセージ）・ネットワークエラー・`userEmail` 未設定（セッション無し、email無し）の6ケースをカバー。
- `useUpdateUserSettingsMutation` の正常更新（`invalidateQueries` 含む）・APIエラー・ネットワークエラー・セッション未設定時の `queryKey: ["user-settings", undefined]` 計6ケースをカバー。前回のレビューで指摘された `invalidateQueries` アサーションの順序も `waitFor` 内に移動されており修正済み。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

テストのみ追加で既存の実装コードに変更なし。マージしても安全。

変更内容は新規テストファイルの追加のみで、フック本体への変更は一切ない。前回の指摘事項（invalidateQueries アサーションの waitFor 外配置・as any キャスト・セッション欠如テストの不在）はすべて解消されており、テストロジックに新たな欠陥は見当たらない。

特になし。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx | 新規テストファイル。12テストで両フックの主要パスを網羅。前回の指摘（invalidateQueriesアサーション順序、as anyキャスト、セッション欠如テスト）は全て解消済み。 |

</details>

</details>

<sub>Reviews (8): Last reviewed commit: ["test: add unit tests for useUserSettings..."](https://github.com/hiroki-org/audicle/commit/d3259993d2a2c98860c2c115a8d7daa747ef77bf) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=31722476)</sub>

### CODE REVIEW

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

File: packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    }
+  });
+
+  const wrapper = ({ children }: { children: React.ReactNode }) => (
+    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+  );
+
+  describe("useUserSettings", () => {
+    it("should fetch user settings successfully", async () => {
+      const mockSettings = { theme: "dark", playbackSpeed: 1.5 };
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx]: Addressed in 9140a8c: mock settings now use the typed UserSettings shape with playback_speed, voice_model, language, and color_theme.

File: packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      const { result } = renderHook(() => useUserSettings(), { wrapper });
+
+      expect(result.current.fetchStatus).toBe("idle");
+      expect(fetchMock).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("useUpdateUserSettingsMutation", () => {
+    it("should update user settings successfully and invalidate queries", async () => {
+      const mockSettings = { theme: "light", playbackSpeed: 1.0 };
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx]: Addressed in 9140a8c: mock settings now use valid UserSettings fields and values.

[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx]: Addressed in 9140a8c: removed the as any casts by typing mockSettings as UserSettings.

File: packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+      result.current.mutate(mockSettings as any);
+
+      await waitFor(() => expect(result.current.isError).toBe(true));
+
+      expect(result.current.error?.message).toBe("Mutation specific error");
+    });
+
+    it("should handle mutation API error with default message", async () => {
+      const mockSettings = { theme: "light", playbackSpeed: 1.0 };
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx]: Addressed in 9140a8c: the test data now mirrors the UserSettings interface exactly.

File: packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+      });
+
+      expect(invalidateQueriesSpy).toHaveBeenCalledWith({
+        queryKey: ["user-settings", "test@example.com"],
+      });
+      expect(result.current.data).toEqual(mockResponse);
+    });
+
+    it("should handle mutation API error with specific message", async () => {
+      const mockSettings = { theme: "light", playbackSpeed: 1.0 };
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx]: Addressed in 9140a8c: the shared mockSettings object is typed as UserSettings and uses the expected snake_case fields.

File: packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        ok: true,
+        json: async () => mockResponse,
+      } as Response);
+
+      const { result } = renderHook(() => useUpdateUserSettingsMutation(), { wrapper });
+
+      result.current.mutate(mockSettings as any);
+
+      await waitFor(() => expect(result.current.isSuccess).toBe(true));
+
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx]: Addressed in 9140a8c: added an unauthenticated mutation test that verifies invalidation with an undefined user email.

File: packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    (useSession as jest.Mock).mockReturnValue(mockSession);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    if (!originalFetch) {
+      delete (global as typeof globalThis & { fetch?: typeof fetch }).fetch;
+    }
+  });
+
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useUserSettings.test.tsx]: Addressed in 9140a8c: wrapper is declared in the describe scope and assigned inside beforeEach after each QueryClient is created.

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
Closing in favor of #799. This Jules-managed branch kept receiving unrelated automation rewrites after review fixes, while #799 carries the same test coverage from a clean main-based branch.
